### DISCUSSION TITLE
How to make texture (clipPath) active and selected from the code ?

### DISCUSSION THREAD

**Discussion by TomasMatusek** [fabricjs/fabric.js#9677] [Q&A]
Hi, I am trying to create simple editor where I put photos on different predefined positions. This photos can be resized and moved around within the given position. I did it in a way that each (gray) position is Rect object added to the main canvas (background) and photos are clipPath's of those Rect objects.

Problem I am facing is that each photo (texture alias Rect's clipPath) can be resized beyond it's Rect area, which is fine, but then they are overlapping photo (texture) above. It's not overlapping image but you can see that selection area (that blue border) is over second photo at the bottom. Issue is that then when I try to click on second photo I can't select it due to this overllap.

I tried to fix this issue in a way that I am listening to mouse:click event. When mouse is clicked over give Rect position I make that Rect's texture active. So basically I am trying to programatically make that texture active. Problem is it is not working as expected. Here is a video sample:

https://github.com/fabricjs/fabric.js/assets/6783891/d8dfc6ab-d0cb-4f19-a37d-c33f6b3a74db

You can also check demo here: [EDITOR](https://vlastny-kalendar.sk/index.php?option=com_calendars&view=editor)

From that recording you can see that when I stretch top image to overlap bottom one, I am not able to do selection of bottom one when bottom Rect is clicked. Event bottom appears to be selected I am still moving top one for some reason. Here is code snippet:

```
canvas.on('mouse:down', function(event) {
    // Logic to select Rect based on mouse coordinates
    const positionClickedRect = EditorUtils.getPositionRectBasedOnMousePosition(
        positionsRect, textures, event.pointer.x, event.pointer.y
    );
    console.log('position clicked', positionClickedRect);

    if (positionClickedRect) {
        // Deselect all textures 
        textures.forEach(texture => {
            canvas.discardActiveObject(texture)
        });
        
        // Make selected only texture that belongs to the clicked Rect
        const texture = textures.find(texture => texture.positionId === positionClickedRect.positionId);
        if (texture) {
            canvas.setActiveObject(texture).renderAll();
            console.log('texture clicked', texture);
            console.log('active object set', canvas.getActiveObject());
        }
    }
});
```

I am thinking of multiple solutions:
1) Fix that object selection as it should work with current version
2) Limit the texture's blue border somehow to not go beyond the Rect area so it is never overlapping other positions beacuse I am anyway planing to add icons for resizing images so those controls can be hidden.

I would appreciate any help. THANKS! :)

**Comment by ShaMan123** [comment]
Basically you want the clip path to define your selected area
That can be done for sure by testing that the point is contained in it by overriding the following:
https://github.com/fabricjs/fabric.js/blob/5ea2db7bff3b3c3e7d577b98d6d780c737b4f0d0/src/canvas/SelectableCanvas.ts#L797-L806


Also you can refer to https://github.com/fabricjs/fabric.js/pull/7879
